KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs

```diff
@@ -173,7 +183,91 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
                 }
             }
 
+            foreach (var property in richTextPropertiesToBeProcessed)
+            {
+                var value = property.GetValue(instance);
+                var propValue = ((JObject)item["elements"]).Properties()
+                    ?.FirstOrDefault(p => PropertyMapper.IsMatch(property, p.Name, system?.Type))
+                    ?.FirstOrDefault()["value"];
+
+                var modularContentInRichText =
+                                ((JObject)propValue?.Parent?.Parent)?.Property("modular_content")?.Value;
+
+                var currentlyProcessedString = new RichTextContentItem()
+                {
+                    ContentItemCodeName = system.Codename,
+                    RichTextElementCodeName = property.Name
+                };
+                if (currentlyResolvedRichStrings.Contains(currentlyProcessedString))
+                {
+                    value = RemoveContentItemsFromRichText((string) value);     //In case we've stumbled upon an element which is already being processed, we need to use it as 
+                                                                                //is (therefore removing content items) to prevent circular dependency
+                }
+                else
+                {
+                    currentlyResolvedRichStrings.Add(currentlyProcessedString);
+                    value = ProcessContentItemsInRichText(modularContent, processedItems, (string) value,
+                        modularContentInRichText, currentlyResolvedRichStrings);
+                    currentlyResolvedRichStrings.Remove(currentlyProcessedString);
+                }
+                if (value != null)
+                {
+                    property.SetValue(instance, value);
+                }
+
+            }
+
             return instance;
         }
+
+        private object ProcessContentItemsInRichText(JToken modularContent, Dictionary<string, object> processedItems, string value, JToken modularContentInRichText, HashSet<RichTextContentItem> currentlyResolvedRichStrings)
```

Comment:
Yes, that is true, it has occured to me in case somebody would want to just prepare data with our resolving and then do actual resolving later (for instance in View of MVC app). And they still can do it if they want by calling Process by themselves on string returned by GetString. We can get back to this once we get data on how are non-strongly typed models used, and how people would like them to be used.